src/components/CreateArticle.jsx

```diff
@@ -0,0 +1,231 @@
+    }
+
+    if (!regexBody.test(articleBody) && articleBody.length < 20) {
+      articleIsValid = false;
+      errors.articleBody = '*Article body must be of length geater than 20';
+    }
+    if (tags !== '') {
+      if (tags.length < 2) {
+        articleIsValid = false;
+        errors.tags = '*Tag must be of length greater than or equal to 2';
+      }
+    }
+    this.setState({
+      errors
+    });
+    return articleIsValid;
+  };
+
+  handleAddition = (tags) => {
+    this.setState({ errors: {}, tags });
+  };
+
+  handleChange = (event) => {
+    const { name, value } = event.target;
+    this.setState({
+      errors: {},
+      [name]: value
+    });
+  };
+
+  handleImageDelete = (event) => {
+    event.preventDefault();
+    this.setState({
+      image: null,
+      displayImage: null
+    });
+  };
+
+  onSubmit = async (event) => {
+    event.preventDefault();
+    const data = { ...this.state };
+    const { tags } = data;
+    data.tags = tags.join();
+
+    if (this.handleArticleValidation(data.articleTitle, data.articleBody, data.tags)) {
+      if (!data.tags) {
+        if (data.image) formData.append('image', data.image);
+        formData.append('articleTitle', data.articleTitle);
+        formData.append('articleBody', data.articleBody);
+        this.props.createUserArticle(formData, this.props);
+      } else {
+        if (data.image) formData.append('image', data.image);
+        formData.append('articleTitle', data.articleTitle);
+        formData.append('articleBody', data.articleBody);
+        formData.append('tags', data.tags);
+        this.props.createUserArticle(formData, this.props);
+      }
+    }
+  };
+
+  render() {
+    const { articleTitle, tags, displayImage } = this.state;
+    const { isLoading } = this.props.articles;
+    return (
+      <div className="text-editor">
+        <form onSubmit={this.onSubmit}>
+          <textarea
+            rows="1"
+            cols="50"
+            name="articleTitle"
+            value={articleTitle}
+            placeholder="Title"
+            onChange={this.handleChange}
+          />
+          <div className="errorMsg">{this.state.errors.articleTitle}</div>
+          <div className="article-button">
+            <div className="image-button">
+              <label htmlFor="image">
+                <i className="fa fa-upload" />{' '}
+                <span>Attach a cover picture (Optional)</span>
+                <input
+                  id="image"
+                  type="file"
+                  image=""
+                  name="image"
+                  className="input"
+                  onChange={this.handleImageUpload}
+                  accept="image/*"
+                />
+              </label>
+            </div>
+            {displayImage ? (
+              <span>
+                <i className="fa fa-times" onClick={this.handleImageDelete} />
+              </span>
+            ) : null}
```

Comment:
Better to short-circuit here:
```suggestion
{displayImage && (
              <span>
                <i className="fa fa-times" onClick={this.handleImageDelete} />
              </span>
            )}
```